sysbuild/CMakeLists.txt

```diff
@@ -52,38 +94,144 @@ function(${SYSBUILD_CURRENT_MODULE_NAME}_pre_cmake)
+    # merged hex file
+    set_target_properties(mcuboot PROPERTIES BUILD_ONLY true)
+    set_property(TARGET mcuboot APPEND_STRING PROPERTY CONFIG "CONFIG_UPDATEABLE_IMAGE_NUMBER=${SB_CONFIG_MCUBOOT_UPDATEABLE_IMAGES}\n")
+    set_property(TARGET ${DEFAULT_IMAGE} APPEND_STRING PROPERTY CONFIG "CONFIG_UPDATEABLE_IMAGE_NUMBER=${SB_CONFIG_MCUBOOT_UPDATEABLE_IMAGES}\n")
+
+    # Set MCUboot operating mode in application and MCUboot
+# todo: support RAM load
+    set(mcuboot_mode)
+    set(application_mcuboot_mode)
+    set(mcuboot_modes "CONFIG_SINGLE_APPLICATION_SLOT;CONFIG_BOOT_SWAP_USING_MOVE;CONFIG_BOOT_SWAP_USING_SCRATCH;CONFIG_BOOT_UPGRADE_ONLY;CONFIG_BOOT_DIRECT_XIP;CONFIG_BOOT_DIRECT_XIP_REVERT")
+    set(application_mcuboot_modes "CONFIG_MCUBOOT_BOOTLOADER_MODE_SINGLE_APP;CONFIG_MCUBOOT_BOOTLOADER_MODE_SWAP_WITHOUT_SCRATCH;CONFIG_MCUBOOT_BOOTLOADER_MODE_SWAP_SCRATCH;CONFIG_MCUBOOT_BOOTLOADER_MODE_OVERWRITE_ONLY;CONFIG_MCUBOOT_BOOTLOADER_MODE_DIRECT_XIP;CONFIG_MCUBOOT_BOOTLOADER_MODE_DIRECT_XIP_WITH_REVERT")
+
+    if(SB_CONFIG_MCUBOOT_MODE_SINGLE_APP)
+      set(mcuboot_mode CONFIG_SINGLE_APPLICATION_SLOT)
+      set(application_mcuboot_mode CONFIG_MCUBOOT_BOOTLOADER_MODE_SINGLE_APP)
+    elseif(SB_CONFIG_MCUBOOT_MODE_SWAP_WITHOUT_SCRATCH)
+      set(mcuboot_mode CONFIG_BOOT_SWAP_USING_MOVE)
+      set(application_mcuboot_mode CONFIG_MCUBOOT_BOOTLOADER_MODE_SWAP_WITHOUT_SCRATCH)
+    elseif(SB_CONFIG_MCUBOOT_MODE_SWAP_SCRATCH)
+      set(mcuboot_mode CONFIG_BOOT_SWAP_USING_SCRATCH)
+      set(application_mcuboot_mode CONFIG_MCUBOOT_BOOTLOADER_MODE_SWAP_SCRATCH)
+    elseif(SB_CONFIG_MCUBOOT_MODE_OVERWRITE_ONLY)
+      set(mcuboot_mode CONFIG_BOOT_UPGRADE_ONLY)
+      set(application_mcuboot_mode CONFIG_MCUBOOT_BOOTLOADER_MODE_OVERWRITE_ONLY)
+    elseif(SB_CONFIG_MCUBOOT_MODE_DIRECT_XIP)
+      set(mcuboot_mode CONFIG_BOOT_DIRECT_XIP)
+      set(application_mcuboot_mode CONFIG_MCUBOOT_BOOTLOADER_MODE_DIRECT_XIP)
+    elseif(SB_CONFIG_MCUBOOT_MODE_DIRECT_XIP_WITH_REVERT)
+      set(mcuboot_mode CONFIG_BOOT_DIRECT_XIP;CONFIG_BOOT_DIRECT_XIP_REVERT)
+      set(application_mcuboot_mode CONFIG_MCUBOOT_BOOTLOADER_MODE_DIRECT_XIP_WITH_REVERT)
+    endif()
+
+    # Apply configuration to MCUboot
+    foreach(mode ${mcuboot_modes})
+      if("${mode}" IN_LIST mcuboot_mode)
+        set_config_bool(mcuboot ${mode} y)
+      else()
+        set_config_bool(mcuboot ${mode} n)
+      endif()
+    endforeach()
+
+    # Apply configuration to application
+    foreach(mode ${application_mcuboot_modes})
+      if("${mode}" IN_LIST application_mcuboot_mode)
+        set_config_bool(${DEFAULT_IMAGE} ${mode} y)
+      else()
+        set_config_bool(${DEFAULT_IMAGE} ${mode} n)
+      endif()
+    endforeach()
+
+    # Check if the board is a non-secure one
+    string(LENGTH "${BOARD}" board_len)
+    math(EXPR board_len "${board_len} - 3")
+    string(SUBSTRING "${BOARD}" ${board_len} 3 ns_board_check)
+
+    if("${ns_board_check}" STREQUAL "_ns")
```

Comment:
Could this be done with `if (CONFIG_TRUSTED_EXECUTION_NONSECURE)` instead? That way we attach less semantics to the name of the board.

Reply:
No